hackIDE/static/hackIDE/js/custom.js

```diff
@@ -411,6 +411,17 @@ $(document).ready(function(){
 							$(".error-message").html(response.run_status.stderr);
 						}
 					}
+					// To handle case when hackerearth api doesn't return a valid response.
```

Comment:
How do you know it succeeds? Did you check how ajax behaves when the server times out? We need to consider that as well.